### ISSUE TITLE
return axes object in plotting functions

### ISSUE THREAD

**Issue by chrisadas** [shap/shap#197]
Currently, dependence_plot(), summary_plot() finishes with
```
if show:
    pl.show()
return
```
which displays the plots, but does not expose the axes object to outside of the function.

Returning the axes object will allow the plot to be further adjusted (changing axes labels, x- and y-limits, change to log_scale, aspect ratio, etc.

One way might be something like this:
<pre>
# added "ax=None" to function arguments
def dependence_plot(ind, shap_values, features, feature_names=None, display_features=None,
                    interaction_index="auto", color="#1E88E5", axis_color="#333333",
dot_size=16, alpha=1, title=None, show=True<b>, ax=None</b>)

...

<b>if ax is None:
    fig, ax = plt.subplots(1, 1)</b>

# change "pl." to "ax." afterward

if interaction_index is not None:
        <b>ax.</b>scatter(xv, s, s=dot_size, linewidth=0, c=features[:, interaction_index], cmap=red_blue,
                   alpha=alpha, vmin=clow, vmax=chigh, norm=color_norm, rasterized=len(xv) > 500)
    else:
        <b>ax.</b>scatter(xv, s, s=dot_size, linewidth=0, color="#1E88E5",
alpha=alpha, rasterized=len(xv) > 500)

# and finally

<b>return ax</b>
</pre>

I will try to submit PR if the idea is sound.

**Comment by slundberg** [comment]
Thanks for the suggestion! If you just want to adjust the axes and such, you can pass `show=False` and then just call `pl.xlim(0,1)` after the plot, finishing with your own `pl.show()`. You could also call `pl.gca()` if you want the axis shap was using.

There could be other cases where passing back the axis is better. If you have some in mind let me know.

**Comment by chrisadas** [comment]
@slundberg that works like a charm. Thank you!

**Comment by alberto-bracci** [comment]
Hi, I am trying to place summary plots on a subplots grid. something like: 4 models, a normal summary and a bar one for each model on each line. 
I tried using ax = plt.gca() after producing the plot and then placing the ax on a grid generated by fig,axes = plt.subplots(4,2) but it's not working at all, the new grid shows empty even if the axes variable contains the generated axes

Also, it seems plt.gca() catches the axes only if I place it before generating the plot. 

I believe adding an ax parameter to the summary_plot function would work better in this use case, as I did a similar thing with dependence plots and it worked like a charm.
